packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+    /** @internal */
+    public _markAllDefinesAsDirty(): void {
+        this._enable(this._isEnabled);
+        this.markAllDefinesAsDirty();
+    }
+
+    /**
+     * Creates a new MeshDebugPluginMaterial
+     * @param material Material to attach the mesh debug plugin to
+     * @param options Options for the mesh debug plugin
+     */
+    constructor(material: PBRBaseMaterial | StandardMaterial, options?: MeshDebugOptions) {
+        options = options ?? {};
+
+        const defines = new MeshDebugDefines();
+        defines.MODE = options.mode ?? defines.MODE;
+        defines.MULTIPLYDEBUG = options.multiplyDebug ?? defines.MULTIPLYDEBUG;
+        super(material, "MeshDebug", 200, defines);
+
+        options.mode = defines.MODE;
+        options.multiplyDebug = defines.MULTIPLYDEBUG;
+        options.shadedDiffuseColor = options.shadedDiffuseColor ?? new Vector3(1, 1, 1);
+        options.shadedSpecularColor = options.shadedSpecularColor ?? new Vector3(0.8, 0.8, 0.8);
+        options.shadedSpecularPower = options.shadedSpecularPower ?? 10;
+        options.wireframeThickness = options.wireframeThickness ?? 0.7;
+        options.wireframeColor1 = options.wireframeColor1 ?? new Vector3(0, 0, 0);
+        options.wireframeColor2 = options.wireframeColor2 ?? new Vector3(0.9, 0.9, 0.9);
+        options.vertexRadius = options.vertexRadius ?? 1.2;
+        options.uvScale = options.uvScale ?? 20;
+        options.uvColor1 = options.uvColor1 ?? new Vector3(1, 1, 1);
+        options.uvColor2 = options.uvColor2 ?? new Vector3(0.5, 0.5, 0.5);
+        options.materialColor = options.materialColor ?? this._getRandomColor();
+        this._options = options;
+        this._mode = this._options.mode!;
+
+        this._enable(this._isEnabled);
+    }
+
+    /**
+     * Get the class name
+     * @returns Class name
+     */
+    public getClassName() {
+        return "MeshDebugPluginMaterial";
+    }
+
+    /**
+     * Prepare the defines
+     * @param defines Mesh debug defines
+     * @param _scene Scene
+     * @param _mesh Mesh associated with material
+     */
+    public prepareDefines(defines: MeshDebugDefines, scene: Scene, mesh: AbstractMesh) {
+        if ((this._mode == MeshDebugMode.VERTICES || this._mode == MeshDebugMode.TRIANGLES) && !mesh.isVerticesDataPresent("initialPass")) {
+            Logger.Warn("For best results with VERTICES or TRIANGLES mode, please use MeshDebugPluginMaterial.tripleTriangles() on mesh.", 1);
+        }
+        defines.MODE = this._mode;
+        defines.MULTIPLYDEBUG = this._options.multiplyDebug!;
+    }
+
+    /**
+     * Get the shader attributes
+     * @param attributes Array of attributes
+     */
+    public getAttributes(attr: string[]) {
+        attr.push("initialPass");
+    }
+
+    /**
+     * Get the shader uniforms
+     * @returns Uniforms
+     */
+    public getUniforms() {
+        return {
+            ubo: [
+                { name: "m_shadedDiffuseColor", size: 3, type: "vec3" },

```

Comment:
In fact, I thought of something more robust than `m_` when I talked about adding a prefix to variable and function names ;)

Something like `mdpmShadedDiffuseColor`, to take the same prefix as above (but feel free to use whatever you like, as long as it's discriminating enough).